### PR TITLE
add step to purge invalid jobs at the end of worker run

### PR DESCRIPTION
**Pull request by ttngu207**

### CODE REVIEW

File: datajoint_utilities/dj_worker/worker.py [hunk 1]
```diff
@@ -203,6 +203,13 @@ def _keep_running(self):
         exceed_max_idled_cycle = 0 < self._max_idled_cycle < self._idled_cycle_count
         return not (exceed_run_duration or exceed_max_idled_cycle)
 
+    def _purge_invalid_jobs(self):
+        for process_type, process, _ in self._processes_to_run:
+            if process_type != "dj_table":
+                continue
+            vmod = self._pipeline_modules[process.database]
+            purge_invalid_jobs(vmod.schema.jobs, process)
```
Comment (dimitri-yatsenko) [inline datajoint_utilities/dj_worker/worker.py]:
```suggestion
            if process_type == "dj_table":
                vmod = self._pipeline_modules[process.database]
                purge_invalid_jobs(vmod.schema.jobs, process)
```

File: datajoint_utilities/dj_worker/worker.py [hunk 2]
```diff
@@ -294,14 +306,12 @@ def _purge_invalid_jobs(JobTable, table):
...
-        for key, job_key in zip(*jobs_query.fetch("KEY", "key")):
-            if not (table.key_source & job_key):
-                (jobs_query & key).delete()
-                invalid_removed += 1
-
-        logger.info(
-            f"{invalid_removed}/{invalid_count} invalid jobs removed for `{dj.utils.to_camel_case(table.table_name)}`"
-        )
+    for key, job_key in zip(*jobs_query.fetch("KEY", "key")):
+        if (not (table.key_source & job_key)) or (table & job_key):
```
Comment (dimitri-yatsenko) [inline datajoint_utilities/dj_worker/worker.py:310]: Does this have lower performance? It definitely reads better.
```suggestion
        if job_key not in table.key_source or job_key in table:
```

Comment (ttngu207) [reply datajoint_utilities/dj_worker/worker.py:310]: yes it does

`table.key_source & job_key` is much faster than `job_key in table.key_source`

Comment (dimitri-yatsenko) [reply datajoint_utilities/dj_worker/worker.py:310]: This does not try to be transaction safe. Is this only used in low-activity scenarios?

Comment (dimitri-yatsenko) [reply datajoint_utilities/dj_worker/worker.py:310]: I see. Because this is just the jobs table, which is not critical for data consistency.

**Review by dimitri-yatsenko — APPROVED** [review]

**Pull request merged**